packages/lab/src/stepped-tracker/Step.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  type ButtonProps,
+  makePrefixer,
+  useControlled,
+  useId,
+} from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import clsx from "clsx";
+import {
+  type CSSProperties,
+  type ComponentProps,
+  type ReactNode,
+  useContext,
+  useEffect,
+} from "react";
+
+import { StepConnector } from "./Step.Connector";
+import { StepDescription } from "./Step.Description";
+import { StepExpandTrigger } from "./Step.ExpandTrigger";
+import { StepIcon } from "./Step.Icon";
+import { StepLabel } from "./Step.Label";
+import stepCSS from "./Step.css";
+import { SteppedTracker } from "./SteppedTracker";
+import { DepthContext } from "./SteppedTracker.Provider";
+
+export namespace Step {
+  export interface Props extends Omit<ComponentProps<"li">, "onToggle"> {
+    id?: string;
+    label?: ReactNode;
+    description?: ReactNode;
+    status?: Step.Status;
+    stage?: Step.Stage;
+    expanded?: boolean;
+    defaultExpanded?: boolean;
+    onToggle?: ButtonProps["onClick"];
+    substeps?: Step.Props[];
+    children?: ReactNode;
+  }
+
+  export type Status = "warning" | "error";
+
+  export type Stage =
+    | "pending"
+    | "locked"
+    | "completed"
+    | "inprogress"
+    | "active";
+
+  /**
+   * noun: elucidation; pl. noun: elucidations
+   * explanation that makes something clear; 🤔
+   */
+  export type Elucidation = Status | Stage;
+
+  export type Depth = number;
+}
+
+const withBaseName = makePrefixer("saltStep");
+
+export function Step({
+  id: idProp,
+  label,
+  description,
+  status,
+  stage = "pending",
+  expanded: expandedProp,
+  defaultExpanded,
+  onToggle,
+  className,
+  style,
+  substeps,
+  children,
+  ...props
+}: Step.Props) {
+  const id = useId(idProp);
+  const targetWindow = useWindow();
+  const depth = useContext(DepthContext);
+
+  const [expanded, setExpanded] = useControlled({
+    controlled: expandedProp,
+    default: Boolean(defaultExpanded),
+
+    name: "Step",
+    state: "expanded",
+  });
+
+  useComponentCssInjection({
+    testId: "salt-step",
+    css: stepCSS,
+    window: targetWindow,
+  });
+
+  useEffect(() => {
+    if (depth === -1) {
+      console.warn(
+        "<Step /> should be used within a <SteppedTracker /> component!",
```

Comment:
only warn in non-prod env, refer to `useControlled`

Reply:
Done!